packages/@atjson/document/src/annotation.ts

```diff
@@ -117,8 +133,10 @@ export default abstract class Annotation<Attributes = {}> {
     let AnnotationClass = this.getAnnotationConstructor();
     let AnnotationToCompareClass = annotationToCompare.getAnnotationConstructor();
 
-    let lhsAnnotationAttributes = this.attributes;
-    let rhsAnnotationAttributes = annotationToCompare.attributes;
+    let lhsAnnotationAttributes =
```

Comment:
Since we're allocating all the keys when removing from undefined, does it make sense to get all keys up front?

Reply:
@tim-evans sorry not sure what you mean exactly. Are you referring to requiring that the keys be passed in as function arguments?

Reply:
Yeah, it looks like we're grabbing all the keys below, does it make sense to do a union between the two sets of keys?

Reply:
hmmm not sure. What do you think would be improved by updating it?

Reply:
Honestly, idk. I think looking at the performance profiles for this may prove to be useful

Reply:
Yea I we'd certainly take a slight perf hit on this change, but I'm currently not seeing how else this is achievable being that we have to remove these undefined variables to ensure equality.